src/dbnode/storage/shard.go

```diff
@@ -1878,6 +1878,29 @@ func (s *dbShard) Bootstrap(
 		s.setFlushStateColdVersion(at, info.VolumeIndex)
 	}
 
+	// May be nil depending on the caching policy.
+	if retrieverMgr := s.opts.DatabaseBlockRetrieverManager(); retrieverMgr != nil {
+		// Now that this shard has finished bootstrapping attempt to cache all of its seekers. Cannot call
+		// this earlier as block lease verification will fail due to the shards not being bootstrapped
+		// (and as a result no leases can be verified since the flush state is not yet known).
+		retriever, err := retrieverMgr.Retriever(s.namespace)
+		if err != nil {
+			multiErr = multiErr.Add(err)
+		} else {
+			s.logger.Debug("caching shard indices",
+				zap.Uint32("shard", s.ID()))
+			if err := retriever.CacheShardIndices([]uint32{s.ID()}); err != nil {
```

Comment:
Heh, I just realized calling CacheShardIndices here will cause parallel caching due to shard.Bootstrap being called at `NumCPU/2` (which honestly is probably desired since it's not 100% disk IO bound and does do parsing, etc).

Will be interesting to see if this speeds up startup speed at all.